docs/resources/iotda_device_linkage_rule.md

```diff
@@ -0,0 +1,282 @@
+The `actions` block supports:
+
+* `type` - (Required, String) Specifies the type of action. The options are as follows:
+  + **DEVICE_CMD**: Deliver commands.
+  + **SMN_FORWARDING**: Send SMN notifications.
+  + **DEVICE_ALARM**: Report alarms or clear alarms.
+
+* `device_command` - (Optional, List) Specifies the detail of device command. It is required when type
+is `DEVICE_CMD`. The [device_command](#IoTDA_device_command) structure is documented below.
+
+* `smn_forwarding` - (Optional, List) Specifies the detail of SMN notifications. It is required when type
+is `SMN_FORWARDING`. The [smn_forwarding](#IoTDA_smn_forwarding) structure is documented below.
+
+* `device_alarm` - (Optional, List) Specifies the detail of device alarm. It is required when type
+is `DEVICE_ALARM`. The [device_alarm](#IoTDA_device_alarm) structure is documented below.
+
+<a name="IoTDA_device_command"></a>
+The `device_command` block supports:
+
+* `device_id` - (Required, String) Specifies the device id which excutes the command.
+
+* `service_id` - (Required, String) Specifies the service id to which the command belongs.
+
+* `command_name` - (Required, String) Specifies the command name.
+
+* `command_body` - (Required, String) Specifies the command parameters, in json format.
+  + Example of device command using LWM2M protocol: `{"value":"1"}`, there are key-value pairs, each key is the
+    parameter name of the command in the product model.
+  + Example of device command using MQTT protocol: `{"header": {"mode": "ACK","from": "/users/testUser","method":
+    "SET_TEMPERATURE_READ_PERIOD","to":"/devices/{device_id }/services/{service_id}"},"body": {"value" : "1"}}`.
+      - **mode**: Required, whether the device needs to reply to the confirmation message after receiving the command.
+        The default is ACK mode. `ACK` indicates that the confirmation message needs to be replied,
+        `NOACK` indicates that the confirmation message does not need to be replied.
+      - **from**: Optional, the address of the command sender.
+        When the App initiates a request, the format is /users/{userId},
+        when the application server initiates a request, the format is /{serviceName},
+        and when the IoT platform initiates a request, the format is /cloud/{serviceName}.
+      - **to**: optional, the address of the command receiver, the format is /devices/{device_id}/services/{service_id}.
+      - **method**: optional, the command name defined in the product model.
+      - **body**: optional, the message body of the command, which contains key-value pairs, each key is the parameter
+        name of the command in the product model. The specific format requires application and device conventions.
+
+<a name="IoTDA_smn_forwarding"></a>
+The `smn_forwarding` block supports:
+
+* `region` - (Required, String) Specifies the region to which the SMN belongs.
+
+* `project_id` - (Required, String) Specifies the project ID to which the SMN belongs.
```

Comment:
can we make the project default to region as what we does in provider configuration

Reply:
OK